tests/test_form.py

```diff
@@ -4,106 +4,53 @@
-            "first_aid_fr": ["Rincer les yeux avec de l'eau."],
-            "guaranteed_analysis": [{"nutrient": "Nitrogen", "percentage": "10"}]
-        }
-
-        form = FertiliserForm(**valid_data)
-        self.assertIsInstance(form, FertiliserForm)
-
-    def test_invalid_npk(self):
-        invalid_data = {
-            "npk": "10-10",
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_percentage(self):
-        invalid_data = {
-            "micronutrients_en": [{"name": "Zinc", "percentage": "five"}],
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_weight_kg(self):
-        invalid_data = {
-            "weight_kg": "50pounds",
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_weight_lb(self):
-        invalid_data = {
-            "weight_lb": "110kg",
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_density(self):
-        invalid_data = {
-            "density": "density",
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_humidity(self):
-        invalid_data = {
-            "specifications_en": [{"humidity": "high", "ph": "6.5", "solubility": "high"}],
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_ph(self):
-        invalid_data = {
-            "specifications_en": [{"humidity": "5", "ph": "acidic", "solubility": "high"}],
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-    def test_invalid_solubility(self):
-        invalid_data = {
-            "specifications_en": [{"humidity": "5", "ph": "6.5", "solubility": "very"}],
-        }
-        with self.assertRaises(ValidationError):
-            FertiliserForm(**invalid_data)
-
-if __name__ == "__main__":
+    def test_valid_fertiliser_form(self):
+        data = {
+            "company_name": "ABC Company",
+            "company_address": "123 Main St",
+            "fertiliser_name": "Super Fertiliser",
+            "npk": "10-5-5",
+            "instructions_en": ["Use as directed"],
+            "micronutrients_en": [{"nutrient": "Iron", "value": "2", "unit": "%"}],
+            "specifications_en": [{"humidity": "Low", "ph": "7", "solubility": "High"}],
+            "guaranteed_analysis": [{"nutrient": "Nitrogen", "value": "10", "unit": "%"}]
+        }
+
+        try:
+            form = FertiliserForm(**data)
+        except ValidationError as e:
+            self.fail(f"Validation error: {e}")
+
+        self.assertEqual(form.company_name, "ABC Company")
```

Comment:
Take a look at https://github.com/ai-cfia/nachet-datastore/blob/9278eda4a03d5256dcbd3d64646c44dbdca75b6e/tests/test_metadata.py#L63 This should make your testing less redundant 😉